src/WrapperApp/components/Simulation/SimulationProgressBar.tsx

```diff
@@ -30,16 +30,21 @@ const statusToColor = (status: StatusState) => {
 	}
 };
 
-export function SimulationProgressBar({ status }: SimulationProgressBarProps) {
-	const updateProgress = useCallback(() => {
-		const progress = (status?.simulatedPrimaries ?? 0) / (status?.requestedPrimaries ?? 1);
+export function SimulationProgressBar({ status }: Readonly<SimulationProgressBarProps>) {
+	const isJobDataInvalid = (jobStatus: any): boolean => {
+		return (
+			(!jobStatus?.simulatedPrimaries && jobStatus.simulatedPrimaries !== 0) ||
+			!jobStatus?.requestedPrimaries
+		);
+	};
 
-		return progress;
+	const progressValue = useMemo(() => {
+		if (isJobDataInvalid(status)) {
+			return 0;
+		}
```

Comment:
Physicist reading that code would need quite some time to get the logic.
The code would be much simpler with opposite logic, like defining `isJobDataValid`

Then we need a comment with user-oriented language like:

```
    // 'simulatedPrimaries' is valid if it's set to any number (including 0).
    // 'requestedPrimaries' is valid only if it's set to a non-zero number.
    ```